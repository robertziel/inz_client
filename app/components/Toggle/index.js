import React from 'react';
import PropTypes from 'prop-types';

import ToggleOption from './ToggleOption';

function Toggle(props) {
  let content = <option>--</option>;

  if (props.values) {
    content = props.values.map(value => (
      <ToggleOption key={value} value={value} message={props.messages[value]} />
    ));
  }

  return (
    <select value={props.value} onChange={props.onToggle}>
      {content}
    </select>
  );
}

Toggle.propTypes = {
  onToggle: PropTypes.func,
  values: PropTypes.array,
  value: PropTypes.string,
  messages: PropTypes.object,
};

export default Toggle;
