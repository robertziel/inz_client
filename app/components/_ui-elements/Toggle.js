import styled from 'styled-components';

import ToggleCore from './ToggleCore';

const Toggle = styled(ToggleCore)``;

export { Toggle };
