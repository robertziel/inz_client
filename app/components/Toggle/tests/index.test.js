import React from 'react';
import { mount } from 'enzyme';
import { IntlProvider, defineMessages } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';

import Toggle from '../index';

const defaultEnMessage = 'someContent';
const defaultPlMessage = 'someOtherContent';

const messages = defineMessages({
  en: {
    id: 'app.components.LocaleToggle.en',
    defaultMessage: defaultEnMessage,
  },
  pl: {
    id: 'app.components..LocaleToggle.en',
    defaultMessage: defaultPlMessage,
  },
});

describe('<Toggle />', () => {
  let store;

  describe('render', () => {
    let wrapper;

    function mountWrapper(values) {
      return mount(
        <Provider store={store}>
          <IntlProvider locale="en">
            <Toggle values={values} messages={messages} />
          </IntlProvider>
        </Provider>,
      );
    }

    beforeAll(() => {
      store = configureStore({}, browserHistory);
    });

    it('should contain default text', () => {
      wrapper = mountWrapper(['en', 'pl']);
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should not have ToggleOptions if props.values is not defined', () => {
      wrapper = mountWrapper(null);
      expect(wrapper.find('option')).toHaveLength(1);
      expect(
        wrapper
          .find('option')
          .first()
          .text(),
      ).toEqual('--');
    });
  });
});
