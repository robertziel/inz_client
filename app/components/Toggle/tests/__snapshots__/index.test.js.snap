// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> render should contain default text 1`] = `"<select><option value=\\"en\\">someContent</option><option value=\\"pl\\">someOtherContent</option></select>"`;
